diff -rupN a/drivers/hwmon/Kconfig b/drivers/hwmon/Kconfig
--- a/drivers/hwmon/Kconfig	2024-10-17 10:15:49.000000000 +0200
+++ b/drivers/hwmon/Kconfig	2024-10-17 14:17:58.025488363 +0200
@@ -2109,6 +2109,16 @@ config SENSORS_SURFACE_TEMP
 	  This driver can also be built as a module. If so, the module
 	  will be called surface_temp.
 
+config SENSORS_SURFACE_TEMP
+	tristate "Microsoft Surface Thermal Sensor Driver"
+	depends on SURFACE_AGGREGATOR
+	help
+	  Driver for monitoring thermal sensors connected via the Surface
+	  Aggregator Module (embedded controller) on Microsoft Surface devices.
+
+	  This driver can also be built as a module. If so, the module
+	  will be called surface_temp.
+
 config SENSORS_ADC128D818
 	tristate "Texas Instruments ADC128D818"
 	depends on I2C
diff -rupN a/drivers/hwmon/surface_temp.c b/drivers/hwmon/surface_temp.c
--- a/drivers/hwmon/surface_temp.c	2024-10-17 10:15:49.000000000 +0200
+++ b/drivers/hwmon/surface_temp.c	2024-10-17 14:17:58.026488371 +0200
@@ -27,7 +27,7 @@
  * zeros after the name, so perhaps they can be longer. This number reflects
  * the maximum zero-padded space observed in the returned buffer.
  */
-#define SSAM_TMP_SENSOR_NAME_LENGTH	18
+#define SSAM_TMP_SENSOR_NAME_LENGTH 18
 
 struct ssam_tmp_get_name_rsp {
 	__le16 unknown1;
@@ -148,6 +148,10 @@ static int ssam_temp_hwmon_read_string(s
 static const struct hwmon_channel_info * const ssam_temp_hwmon_info[] = {
 	HWMON_CHANNEL_INFO(chip,
 			   HWMON_C_REGISTER_TZ),
+	/*
+	 * We have at most SSAM_TMP_SENSOR_MAX_COUNT = 16 thermal sensor
+	 * channels.
+	 */
 	HWMON_CHANNEL_INFO(temp,
 			   HWMON_T_INPUT | HWMON_T_LABEL,
 			   HWMON_T_INPUT | HWMON_T_LABEL,
