diff -rupN a/drivers/platform/surface/surface3-wmi.c b/drivers/platform/surface/surface3-wmi.c
--- a/drivers/platform/surface/surface3-wmi.c	2025-06-19 16:24:58.000000000 +0200
+++ b/drivers/platform/surface/surface3-wmi.c	2025-06-19 20:27:19.847027009 +0200
@@ -37,6 +37,13 @@ static const struct dmi_system_id surfac
 			DMI_MATCH(DMI_PRODUCT_NAME, "Surface 3"),
 		},
 	},
+	{
+		.matches = {
+			DMI_MATCH(DMI_BIOS_VENDOR, "American Megatrends Inc."),
+			DMI_MATCH(DMI_SYS_VENDOR, "OEMB"),
+			DMI_MATCH(DMI_PRODUCT_NAME, "OEMB"),
+		},
+	},
 #endif
 	{ }
 };
diff -rupN a/sound/soc/codecs/rt5645.c b/sound/soc/codecs/rt5645.c
--- a/sound/soc/codecs/rt5645.c	2025-06-19 16:24:58.000000000 +0200
+++ b/sound/soc/codecs/rt5645.c	2025-06-19 20:27:19.849087001 +0200
@@ -3792,6 +3792,15 @@ static const struct dmi_system_id dmi_pl
 		.driver_data = (void *)&intel_braswell_platform_data,
 	},
 	{
+		.ident = "Microsoft Surface 3",
+		.matches = {
+			DMI_MATCH(DMI_BIOS_VENDOR, "American Megatrends Inc."),
+			DMI_MATCH(DMI_SYS_VENDOR, "OEMB"),
+			DMI_MATCH(DMI_PRODUCT_NAME, "OEMB"),
+		},
+		.driver_data = (void *)&intel_braswell_platform_data,
+	},
+	{
 		/*
 		 * Match for the GPDwin which unfortunately uses somewhat
 		 * generic dmi strings, which is why we test for 4 strings.
diff -rupN a/sound/soc/intel/common/soc-acpi-intel-cht-match.c b/sound/soc/intel/common/soc-acpi-intel-cht-match.c
--- a/sound/soc/intel/common/soc-acpi-intel-cht-match.c	2025-06-19 16:24:58.000000000 +0200
+++ b/sound/soc/intel/common/soc-acpi-intel-cht-match.c	2025-06-19 20:27:19.850137959 +0200
@@ -27,6 +27,14 @@ static const struct dmi_system_id cht_ta
 			DMI_MATCH(DMI_PRODUCT_NAME, "Surface 3"),
 		},
 	},
+	{
+		.callback = cht_surface_quirk_cb,
+		.matches = {
+			DMI_MATCH(DMI_BIOS_VENDOR, "American Megatrends Inc."),
+			DMI_MATCH(DMI_SYS_VENDOR, "OEMB"),
+			DMI_MATCH(DMI_PRODUCT_NAME, "OEMB"),
+		},
+	},
 	{ }
 };
 
